### PR TITLE
DEV-3135: Add state label configurability

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Updates the contributor address block editor so that it can take on additional labels.

#### Why are we doing this? How does it help us?

Better UX for non-USians.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Edit a contribution page and the Contribution Address block.
2. Check the Province checkbox and save changes.
3. Confirm that the field now shows as "State/Province" in the published page and that state is saved properly in a contribution.
4. Edit the page/block.
5. Check the Region checkbox and save changes.
6. Confirm that that the field now shows as "State/Province/Region" in the published page and that state is saved properly.
7. Create a new contribution page.
8. Confirm that the page only shows "State" as the label in the contributor address block.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

We probably want to document this new functionality.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

Mention this to international orgs, perhaps?

#### Are there any smells or added technical debt to note?

Hit a very strange test failure that only happened in CI that doesn't appear related to this PR. It might just be the phase of the moon but I got it to work in c1fc4caa3bfa6da4e5e322ea83fdd0a121c575c8. I don't understand why it failed.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3135

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/src/hooks/useContributionPage/useContributionPage.types.ts [hunk 1]
```diff
@@ -96,6 +97,53 @@ export interface ContributionPageElement {
...
+  content: {
+    /**
+     * Show a "Province" label on the State field?
+     */
+    showProvinceOnStateField?: boolean;
+    /**
+     * Show a "Region" label on the State field?
+     */
+    showRegionOnStateField?: boolean;
+  };
```
Comment (benjaminEwhite) [inline spa/src/hooks/useContributionPage/useContributionPage.types.ts:110]: If using this approach with booleans, seems liked we'd want one for "state" label as well.

We could also just have the backend supply the right labels, in which case, the interface would expect an array of 1 or more enumerated string values.

Comment (nrh-cklimas) [reply spa/src/hooks/useContributionPage/useContributionPage.types.ts:110]: I think we'd at least want booleans here for localization reasons, so the UI could translate values as need be. (If we already had localization set up we could at least use localization keys, but we haven't thought that far yet.)

I'm not sure if we envision ever allowing people to disable the State label, but I'll ask for some cliarification.

**Review by GuiMend — COMMENTED** [review]
Think the changes seem in the right direction. Could you update this PR with the base branch? It's showing additional changes that were added in the previous branch.

File: spa/src/hooks/useContributionPage/useContributionPage.types.ts (hunk 1 above)
Comment (GuiMend) [reply spa/src/hooks/useContributionPage/useContributionPage.types.ts:110]: I think this is a good enough approach for what we need, but as you raised localization @nrh-cklimas, would we have a word for all 3 options in all supported languages? Probably good to handle the scenario where one of those (even when selected) could be undefined/null - or maybe handle this when time comes.

**Review by GuiMend — COMMENTED** [review]
I'm good with this approach 👍

**Review by GuiMend — APPROVED** [review]
👍

**Review by x110dc — APPROVED** [review]

**Pull request merged**